include/measurement_kit/nettest.hpp

```diff
@@ -0,0 +1,1922 @@
+class DnsInjectionSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "DnsInjection";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "FacebookMessenger" nettest.
+class FacebookMessengerSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "FacebookMessenger";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "HttpHeaderFieldManipulation" nettest.
+class HttpHeaderFieldManipulationSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "HttpHeaderFieldManipulation";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "HttpInvalidRequestLine" nettest.
+class HttpInvalidRequestLineSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "HttpInvalidRequestLine";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "MeekFrontedRequests" nettest.
+class MeekFrontedRequestsSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "MeekFrontedRequests";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "MultiNdt" nettest.
+class MultiNdtSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "MultiNdt";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "Ndt" nettest.
+class NdtSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "Ndt";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "TcpConnect" nettest.
+class TcpConnectSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "TcpConnect";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "Telegram" nettest.
+class TelegramSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "Telegram";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "WebConnectivity" nettest.
+class WebConnectivitySettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "WebConnectivity";
+    /* No nettest-specific settings */
+};
+
+/// Settings of the "Whatsapp" nettest.
+class WhatsappSettings : public CommonSettings {
+  public:
+    /// Unique name of this nettest.
+    static constexpr const char *name = "Whatsapp";
+    
+    /// The "all_endpoints" setting of this nettest.
+    bool all_endpoints = false;
+};
+
+} // namespace settings
+
+// Contains the routers
+namespace routers {
+
+/// Routes nettest events to virtual methods.
+class Router {
```

Comment:
This class is really designed to be SWIG friendly. I have separated events routing and managing the nettest lifecycle because it occurred to me that event routing can have a longer lifecycle than just a single nettest.

One potential issue is that the previous API used lambdas rather than virtual methods. However, virtual methods are much more easily wrappable using SWIG. So, virtual methods. However, with a little more templating it is possible to define a derived class of router than takes one or more lambda per events. I am not planning on adding this class now, but I will do that next, because we use lambdas on iOS.